src/dscanner/analysis/redundant_storage_class.d

```diff
@@ -0,0 +1,101 @@
+// Copyright (c) 2018, dlang-community
+// Distributed under the Boost Software License, Version 1.0.
+//    (See accompanying file LICENSE_1_0.txt or copy at
+//          http://www.boost.org/LICENSE_1_0.txt)
+
+module dscanner.analysis.redundant_storage_class;
+
+import std.stdio;
+import std.string;
+import dparse.ast;
+import dparse.lexer;
+import dscanner.analysis.base;
+import dscanner.analysis.helpers;
+import dsymbol.scope_ : Scope;
+
+/**
+ * Checks for redundant storage classes such immutable and __gshared, static and __gshared
+ */
+class RedundantStorageClassCheck : BaseAnalyzer
+{
+	alias visit = BaseAnalyzer.visit;
+	enum string REDUNDANT_VARIABLE_ATTRIBUTES = "Variable declaration for `%s` has redundant attributes (%-(`%s`%|, %)).";
+
+	this(string fileName, bool skipTests = false)
+	{
+		super(fileName, null, skipTests);
+	}
+
+	override void visit(const Declaration node)
+	{
+		checkAttributes(node);
+		node.accept(this);
+	}
+
+	void checkAttributes(const Declaration node)
+	{
+		if (node.variableDeclaration !is null && node.attributes !is null)
+			checkVariableDeclaration(node.variableDeclaration, node.attributes);
+	}
+
+	void checkVariableDeclaration(const VariableDeclaration vd, const Attribute[] attributes)
+	{
+		import std.algorithm.comparison : among;
+		import std.algorithm.searching: all;
+
+		string[] globalAttributes;
+		foreach (attrib; attributes)
+		{
+			if (attrib.attribute.type.among(tok!"shared", tok!"static", tok!"__gshared", tok!"immutable"))
+				globalAttributes ~= attrib.attribute.type.str;
+		}
+		if (globalAttributes.length > 1)
+		{
+			if (globalAttributes.length == 2 &&
+					globalAttributes.all!(a => a.among("shared", "static")) ||
```

Comment:
D-Scanner warning is right here, you should put line 55 and 56 between parens

Reply:
-> https://wiki.dlang.org/Operator_precedence

but I used parens here now for clarity and to satisfy the self-linting.